pkg/controller/elasticsearch/license/apply.go

```diff
@@ -54,7 +65,27 @@ func applyLinkedLicense(
 	if err != nil {
 		return pkgerrors.Wrap(err, "no valid license found in license secret")
 	}
-	return updater(lic)
+
+	err = updater(lic)
+	if err != nil {
+		return err
+	}
+
+	// Store that the trial license has been started in an annotation
+	if isTrial(&lic) {
+		if license.Annotations == nil {
+			license.Annotations = map[string]string{}
+		}
+		license.Annotations[common_license.TrialLicenseStartedAnnotation] = "true"
+
+		err = c.Update(&license)
```

Comment:
IIUC we are duplicating the state of the license here. If there's a problem while setting the annotation (timeout, any K8S API error...), the operator may try to apply the trial license a second time, which will generate an error _(and I think the operator will never be able to reconcile)_.

I think this concern is related to the ones we have with the [blocking calls on the ES API](https://github.com/elastic/cloud-on-k8s/issues/1161).

I just want to be sure that we share the tradeoffs here between keeping a local state vs getting the trial status from the api (like what is done [in the observer](https://github.com/elastic/cloud-on-k8s/blob/master/pkg/controller/elasticsearch/observer/state.go#L55))

Reply:
Agreed. I think even with the observer state there is the danger of running into this issue, as the observed state usually lags behind the actual state. I am thinking no matter what we use to keep the current trial license state (annotation on the secret or observer in-memory state) we probably need to the make the license updater function resilient against a repeated trial activation request. I.e. it needs to be able to deal with this response: 
```
HTTP/1.1 403 Forbidden
content-type: application/json; charset=UTF-8
content-length: 112

{"acknowledged":true,"trial_was_started":false,"error_message":"Operation failed: Trial was already activated."}⏎
```

So maybe the best option is as @barkbay seems to indicate to just do two requests in case of a trial license, fetch the current license and only then try to activate the trial. Even that is not watertight though as another client might start a trial between the `GET` and the `POST` but at least we would be able to automatically recover from this situation.

Reply:
You are absolutely right. I made the trial start more resilient.